src/nwb_datajoint/common/common_artifact.py

```diff
@@ -174,28 +156,57 @@ def _get_artifact_times(recording, zscore_thresh=None, amplitude_thresh=None,
             np.sum(np.logical_or(above_z, above_a), axis=0) >= nelect_above))
     
     valid_timestamps = recording.get_times()
+    above_thresh_times = valid_timestamps[above_thresh] # find timestamps of initial artifact threshold crossings
     
-    # keep track of artifact detected timestamps - this could be saved if we wanted?
-    # artifact_detected_times = valid_timestamps[above_thresh] # the specific artifact times, NOT including the window around them
-
-    # keep track of all the artifact times within each artifact removal window
+    # keep track of all the artifact timestamps within each artifact removal window and the indices of those timestamps
     artifact_times = []
-    for a in above_thresh:
-        if (a - half_window_points) < 0: # make sure indices are valid for the slice
-            a_times = np.copy(valid_timestamps[0:int(a + half_window_points)])
-            artifact_times.append(a_times)
-        else:
-            a_times = np.copy(valid_timestamps[int(a - half_window_points):int(a + half_window_points)])
-            artifact_times.append(a_times)
-    artifact_times = np.asarray(artifact_times)
-    # alternatively, we could store these as intervals instead of arrays of all the times? (using interval_list_union many times or something?)
+    artifact_indices = []
+    for a in above_thresh_times:
+        a_times = np.copy(valid_timestamps[(valid_timestamps > (a - half_removal_window_s)) & (valid_timestamps <= (a + half_removal_window_s))])
+        a_indices = np.argwhere((valid_timestamps > (a - half_removal_window_s)) & (valid_timestamps <= (a + half_removal_window_s)))
+        artifact_times.append(a_times)
+        artifact_indices.append(a_indices)
+    all_artifact_times = reduce(np.union1d, artifact_times)
+    all_artifact_indices = reduce(np.union1d, artifact_indices)
+    # turn artifact detected times into intervals
+    artifact_intervals = get_valid_intervals(all_artifact_times, recording.get_sampling_frequency(), 1.5, .000001)
```

Comment:
Will take a look at this soon, as well as (just making a note here) adding the parameter set 'none' as part of insert_default in case folks want to bypass artifact detection, and considering what the default parameters should really be. Planning to make new PR for that when ready